Code/Tools/ProjectManager/Source/ProjectsScreen.cpp

```diff
@@ -682,13 +682,54 @@ namespace O3DE::ProjectManager
 
         if (foundButton != m_projectButtons.end())
         {
-            (*foundButton).second->SetState(ProjectButtonState::Downloading);
+            (*foundButton).second->SetState(queueBuild ? ProjectButtonState::DownloadingBuildQueued : ProjectButtonState::Downloading);
         }
     }
 
-    void ProjectsScreen::HandleDownloadResult(const QString& /*projectName*/, bool /*succeeded*/)
+    void ProjectsScreen::HandleDownloadResult(const QString& projectName, bool succeeded)
     {
-        ResetProjectsContent();
+        auto foundButton = AZStd::ranges::find_if(
+            m_projectButtons,
+            [&projectName](const AZStd::unordered_map<AZ::IO::Path, ProjectButton*>::value_type& value)
+            {
+                return (value.second->GetProjectInfo().m_projectName == projectName);
+            });
+
+        if (foundButton != m_projectButtons.end())
+        {
+            if (succeeded)
+            {
+                // Find the project info since it should now be local
+                auto projectsResult = PythonBindingsInterface::Get()->GetProjects();
+                if (projectsResult.IsSuccess() && !projectsResult.GetValue().isEmpty())
+                {
+                    for (const ProjectInfo& projectInfo : projectsResult.GetValue())
+                    {
+                        if (projectInfo.m_projectName == projectName)
```

Comment:
All of this assumes the project name is unique.  If a user wanted to download a copy of a project or a project with the same name but different version - we'll need to update this somehow to identify the project button using something unique like project_path

Reply:
I think a larger change to the DownloadController to use AZ::Event or similar instead of the signal/slots would probably help this too, along with not having to check whether the signals are for projects or not. Should probably spend the time during that change to fix the other parts of Project Manager that are also checking projects by name.